packages/contractkit/src/utils/web3-utils.ts

```diff
@@ -71,3 +71,68 @@ class WrappingSubprovider extends Subprovider {
     })
   }
 }
+
+// tslint:disable-next-line:max-classes-per-file
+export class CeloProvider implements Provider {
+  private static createNewProviderWithLocalAccount(existingProvider: Provider): Web3ProviderEngine {
+    // Create a Web3 Provider Engine
+    const providerEngine = new Web3ProviderEngine()
+    // Use the existing provider to route all other requests
+    const wrappingSubprovider = new WrappingSubprovider(existingProvider)
+    debug('Setting up providers...')
+    providerEngine.addProvider(wrappingSubprovider)
+    return providerEngine
+  }
+
+  private web3ProviderEngine: Web3ProviderEngine
+  private signingProviders: CeloPrivateKeyWalletProvider[]
+  private signingAddresses: string[]
+
+  constructor(readonly existingProvider: Provider) {
+    this.web3ProviderEngine = CeloProvider.createNewProviderWithLocalAccount(existingProvider)
```

Comment:
maybe it's me, style: a bit weird to call a private static method from a constructor.